File: Go/003_Program/main.go
package main

import (
    "bufio"
    "fmt"
    "os"
)

func main() {
    fmt.Println("=== File Reader ===")

    var filename string
    fmt.Print("Enter filename: ")
    fmt.Scanln(&filename)

    file, err := os.Open(filename)
    if err != nil {
        fmt.Printf("Error: %v\n", err)
        return
    }
    defer file.Close()

    scanner := bufio.NewScanner(file)
    lineNum := 1

    fmt.Println("\nFile contents:")
    fmt.Println("---")

    for scanner.Scan() {
        fmt.Printf("%3d | %s\n", lineNum, scanner.Text())
        lineNum++
    }

    if err := scanner.Err(); err != nil {
        fmt.Printf("Error reading file: %v\n", err)
    }

    fmt.Printf("\nTotal lines: %d\n", lineNum-1)
}

File: Rust/003_Program/main.rs
use std::fs::File;
use std::io::{self, BufRead, BufReader, Write};

fn main() {
    println!("=== File Reader ===");

    print!("Enter filename: ");
    io::stdout().flush().unwrap();

    let mut filename = String::new();
    io::stdin().read_line(&mut filename).unwrap();
    let filename = filename.trim();

    match File::open(filename) {
        Ok(file) => {
            let reader = BufReader::new(file);
            println!("\nFile contents:");
            println!("---");

            for (line_num, line) in reader.lines().enumerate() {
                match line {
                    Ok(content) => println!("{:3} | {}", line_num + 1, content),
                    Err(e) => eprintln!("Error reading line: {}", e),
                }
            }
        }
        Err(e) => eprintln!("Error opening file: {}", e),
    }
}
